config/initializers/warden.rb

```diff
@@ -25,34 +25,34 @@ def valid?
 
   def authenticate!
     user = User.find_by_email(user_params[:email])
-    if user && user.authenticate(user_params[:password])
+    if user && user.authenticate(user_params[:password]) && user.confirmation_token.nil?
```

Comment:
here if you make that method `confirmed?` that i told you about you can write this as `if user && user.authenticate(user_params[:password]) && user.confirmed?` and it's going to be good.